pkg/runtime/config.go

```diff
@@ -52,6 +59,16 @@ const (
 	DefaultAppHealthCheckPath = "/health"
 )
 
+// IsHTTP returns true if the protocol is HTTP.
+func (p Protocol) IsHTTP() bool {
```

Comment:
nit:
The function sounds like it's checking for protocol == 'HTTP' only.

```suggestion
func (p Protocol) IsGRPCBased() bool {
```

```suggestion
func (p Protocol) IsOverGRPC() bool {
```
Since gRPC also works over HTTP, to avoid confusions maybe change to above and invert the if condition everywhere?

Reply:
I will expand the comment to clarify that it includes HTTP variants. I am not worried about the fact that gRPC is based on HTTP as in this context it should not be cause of confusion.